packages/core/test/insured-bridge/e2e/helpers/WatcherHelpers.js

```diff
@@ -0,0 +1,76 @@
+const exec = require("child_process").exec;
+
+const { getMessagesAndProofsForL2Transaction } = require("@eth-optimism/message-relayer");
+const { predeploys } = require("@eth-optimism/contracts");
+
+const { getProviderUrls } = require("./ArtifactsHelper");
+const { OVM_STATE_COMMITMENT_CHAIN } = require("./OptimismConstants");
+
+async function waitForL1ToL2Transaction(transactionObject, watcher) {
+  await transactionObject.wait(); // Wait for tx on L1 to be included.
+  const [msgHash] = await watcher.getMessageHashesFromL1Tx(transactionObject.hash); // Get msgHash from watcher on L1.
+  await watcher.getL2TransactionReceipt(msgHash, true); // Wait for the hash to be included on L2.
+}
+
+async function waitForL2ToL1Transaction(transactionObject, watcher) {
+  await transactionObject.wait(); // Wait for tx on L2 to be included.
+  const [msgHash] = await watcher.getMessageHashesFromL2Tx(transactionObject.hash); // Get msgHash from watcher on L2.
+  await watcher.getL1TransactionReceipt(msgHash); // Wait for the hash to be included on L1.
+}
+
+// The methods below were unfortunately not used in the e2e tests as the docker contains are somewhat flaky when turning
```

Comment:
I could not get these to work consistently, unfortunately. We can remove them from this PR but I'd like to perhaps use them in the future if we can get the container setup to be more stable.